sdk/storage/storage-blob/test/containerclient.spec.ts

```diff
@@ -226,12 +222,44 @@ describe("ContainerClient", () => {
     }
   });
 
+  it("Verify AsyncIterator for listBlobsFlat", async () => {
+    const blobClients = [];
+    const prefix = "blockblob";
+    const metadata = {
+      keya: "a",
+      keyb: "c"
+    };
+    for (let i = 0; i < 2; i++) {
+      const blobClient = containerClient.createBlobClient(getUniqueName(`${prefix}/${i}`));
+      const blockBlobClient = blobClient.createBlockBlobClient();
+      await blockBlobClient.upload("", 0, {
+        metadata
+      });
+      blobClients.push(blobClient);
+    }
+
+    const iterator = await containerClient.listBlobsFlat({
```

Comment:
Do we need test against other usage scenarios for async iterator? such as `async for loop`

Reply:
Can we cover the cross segment scenario? Because implementation of `listBlobsFlat` needs to handle `marker` update

Reply:
> Can we cover the cross segment scenario? Because implementation of listBlobsFlat needs to handle marker update, while existing test case doesn't cover that

I'm not sure if we are doing it.

https://github.com/Azure/azure-sdk-for-js/blob/bc932b3c4b5a6ddc755f7d74e4c46ad8885d3101/sdk/storage/storage-blob/src/ContainerClient.ts#L630-L634

The current implementation of `listBlobsFlat` doesn't take `marker` as an input.
We can make `marker` as an optional attribute to the method. I'm just not sure if we want to do it.

Reply:
- Sorry for the confusion.
- After an offline discussion with Jeremy, he suggested that you might be referring to the usage of more than one `listBlobFlatSegment` in the same `listBlobsFlat`.
---------------------------------------------
842de33 - Added a for-loop test case.

Outputs (console.logs) to verify your question.
 [By passing `maxresults` in the options for `listBlobsFlat` ]

Case 1 - `maxresults` is not set
```bash
------------------------- called listBlobFlatSegment -------------------------
blockblob/0155925458974007856
blockblob/1155925459003608508
blockblob/2155925459032203396
blockblob/3155925459061100420
    √ Verify AsyncIterator(for-loop syntax) for listBlobsFlat (2714ms)
```

Case 2 - `maxresults` is set to `2`
```bash
------------------------- called listBlobFlatSegment -------------------------
blockblob/0155925451934305295
blockblob/1155925451964308353
------------------------- called listBlobFlatSegment -------------------------
blockblob/2155925451993002640
blockblob/3155925452022007968
    √ Verify AsyncIterator(for-loop syntax) for listBlobsFlat (3012ms)
```